nbconvert/exporters/pdf.py

```diff
@@ -153,42 +163,64 @@ def log_error(command, out):
             self.log.debug(u"%s output: %s\n%s", command[0], command, out)
 
-        return self.run_command(self.bib_command, filename, 1, log_error)
-    
+        return self.run_command(self.bib_command, filename, 1, log_error,
+            raise_on_failure)
+
+    @contextmanager
+    def _fake_output_files_dir(self, resources):
+        cwd = os.path.abspath(os.getcwd())
+        # Map the "working directory" to be the absolute location of the file being processed
+        # This is passed into pandoc conversions so it can convert relative paths to absolute paths
+        resources['working_directory'] = os.path.join(cwd, resources['metadata']['path'] or '')
+        with TemporaryWorkingDirectory() as td:
+            # We need to give safe output paths for latex and pdf conversion tools
+            # This means temporarily internal paths to point to a temp directory,
+            # as the xelatex and bibtex commands can't handle spaces or unicode
+            # characters in paths...
+            true_output_files_dir = resources.get('output_files_dir', cwd)
+            resources['output_files_dir'] = os.path.join(td, 'output_files')
+
+            try:
+                yield
+            finally:
+                resources['output_files_dir'] = true_output_files_dir
+                del resources['working_directory']
+
     def from_notebook_node(self, nb, resources=None, **kw):
-        latex, resources = super(PDFExporter, self).from_notebook_node(
-            nb, resources=resources, **kw
-        )
-        # set texinputs directory, so that local files will be found
-        if resources and resources.get('metadata', {}).get('path'):
-            self.texinputs = resources['metadata']['path']
-        else:
-            self.texinputs = getcwd()
-        
-        self._captured_outputs = []
-        with TemporaryWorkingDirectory():
+        if not resources:
+            resources = {}
+        with self._fake_output_files_dir(resources):
+            latex, resources = super(PDFExporter, self).from_notebook_node(
+                nb, resources=resources, **kw
+            )
+            # set texinputs directory, so that local files will be found
+            if resources and resources.get('metadata', {}).get('path'):
+                self.texinputs = resources['metadata']['path']
+            else:
+                self.texinputs = getcwd()
+
+            self._captured_outputs = []
             notebook_name = 'notebook'
             tex_file = self.writer.write(latex, resources, notebook_name=notebook_name)
             self.log.info("Building PDF")
-            rc = self.run_latex(tex_file)
-            if rc:
-                rc = self.run_bib(tex_file)
-            if rc:
-                rc = self.run_latex(tex_file)
-            
+            self.run_latex(tex_file)
+            # This can fail for non-conversion blocking reasons
+            if self.run_bib(tex_file):
+                self.run_latex(tex_file)
```

Comment:
Technically — if we're running bibTeX, we should be running LaTeX one more time. That is it should look like
```
xelatex file.tex
bibtex file.tex
xelatex file.tex
xelatex file.tex
```

For those who want to know why — https://tex.stackexchange.com/questions/53235/why-does-latex-bibtex-need-three-passes-to-clear-up-all-warnings explains it. But basically because LaTeX was designed in a way that it kept little in memory, it needs to separately create references (LaTeX  run 1) , create a file to house all the bibliography references (bibTeX), add the reference file contents to the page (LaTeX run 2), resolve all references (LaTeX run 3).

Reply:
I believe the ```run_latex``` command actually runs latex 3 times by default, so it should be fine.